src/mumble/TalkingUI.cpp

```diff
@@ -0,0 +1,231 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "TalkingUI.h"
+#include "ClientUser.h"
+#include "Channel.h"
+
+#include <QGroupBox>
+#include <QLabel>
+#include <QGridLayout>
+#include <QVBoxLayout>
+#include <QGuiApplication>
+#include <QtCore/QDateTime>
+#include <QtCore/QTimer>
+#include <QScreen>
+
+// We define a global macro called 'g'. This can lead to issues when included code uses 'g' as a type or parameter name (like protobuf 3.7 does). As such, for now, we have to make this our last include.
+#include "Global.h"
+
+TalkingUI::TalkingUI() {
+	setupUI();
+
+	setWindowTitle(QObject::tr("Talking UI"));
+
+	setAttribute(Qt::WA_ShowWithoutActivating);
+	setWindowFlags(Qt::Window | Qt::WindowStaysOnTopHint);
+
+	m_talkingIcon = QIcon(QLatin1String("skin:talking_on.svg"));
+	m_passiveIcon = QIcon(QLatin1String("skin:talking_off.svg"));
+	m_shoutingIcon = QIcon(QLatin1String("skin:talking_alt.svg"));
+	m_whisperingIcon = QIcon(QLatin1String("skin:talking_whisper.svg"));
+}
+
+void TalkingUI::setupUI() {
+	QVBoxLayout *layout = new QVBoxLayout;
+
+	setLayout(layout);
+}
+
+void TalkingUI::removeUser(unsigned int session) {
+	Entry entry = m_entries[session];
```

Comment:
```suggestion
	const auto iter = m_entries.find(session);
	if (iter == m_entries.constEnd()) {
		return;
	}
	
	auto entry = iter.value();
```

From https://doc.qt.io/qt-5/qhash.html#operator-5b-5d:
> If the hash contains no item with the _key_, the function inserts a [default-constructed value](https://doc.qt.io/qt-5/containers.html#default-constructed-value) into the hash with the _key_, and returns a reference to it. If the hash contains multiple items with the _key_, this function returns a reference to the most recently inserted value.

Reply:
Yeah that's probably a good change. I was just about to say that one can expect the entry for the given UI to exist, but for this specific function this doesn't hold as the failsafe code block in `on_talkingStateChanged` also calls this function.

Good catch! :)

Reply:
No problem!